app/consolidated-view-poc/react/app/useQuery.js

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+import { useToken } from './AuthProvider.js'
+
+export function useLazyQuery(query, { headers, preloaded = null }) {
+  const [loading, setLoading] = useState(false)
+  const [data, setData] = useState(preloaded)
+  const [error, setError] = useState(null)
+  const [preloadedReturned, setPreloadedReturned] = useState(!preloaded)
+  const { getToken } = useToken()
+
+  return [
+    async (variables) => {
+      if (preloadedReturned) {
+        setLoading(true)
+        const token = await getToken()
+
+        try {
+          const response = await fetch('/consolidated-view/graphql', {
+            method: 'POST',
+            headers: {
+              'content-type': 'application/json',
+              Authorization: `Bearer ${token}`,
+              ...headers
+            },
+            body: JSON.stringify({
+              query,
+              variables
+            })
+          })
+          const { data } = await response.json()
+          setData(data)
```

Comment:
I guess there could be errors here too!?
```suggestion
          const { data, errors } = await response.json()
          setData(data)
          if (errors) {
            setError(errors)
          }
```